VsViewer/vs_calc/VsProfile.py

```diff
@@ -54,9 +51,14 @@ def __init__(
                     vs[first_remove] = vs[last_to_keep]
                     vs_sd[first_remove] = vs_sd[last_to_keep]
         self.max_depth = reduce_to
+        if average_vs_under_3m:
+            shal_vs_avg = np.average(vs[(depth >= 2.5) & (depth <= 3.5)])
+            vs[depth <= 3.0] = shal_vs_avg
         self.vs = vs[int_depth_mask]
         self.vs_sd = vs_sd[int_depth_mask]
         self.depth = depth[int_depth_mask]
+        self.eff_stress = None if eff_stress is None else eff_stress[int_depth_mask]
```

Comment:
Quick question, what happens with this line? If eff_stress is None then it remains None? When is that the case?

Reply:
Correct this is called a ternary operation and yes it will stay None if eff_stress is None. This happens when you already have a "VsProfile" file (don't have any examples of real ones, have only self generated) But from what I am aware these files only contain the Vs, Vs_SD and Depth values. Otherwise when coming from a CPT or SPT this will exist.